### PR TITLE
Add missing type information for v3.2 of react-hooks-testing-library 

### LINKED ISSUE #246: Type definitions are not up to date
- `react-hooks-testing-library` version: 3.2.1
- `react-test-renderer` version: 16.12.0
- `react` version: 16.12.0
- `node` version: 12.13.1
- `npm` version: 6.12.1

### Relevant code or config:

```js
let your = (code, tell) => `the ${story}`
```

### What you did:

Using Typescript:
`const { wait } = renderHook(() =>`

### What happened:

I get this error `'wait' is declared but its value is never read.`

### Problem description:

type definitions is no up to date

### Suggested solution:

Somehow update type definition because the current definition for the v3.2.1 is
```javascript
// Type definitions for @testing-library/react-hooks 3.1
// Project: https://github.com/testing-library/react-hooks-testing-library
// Definitions by: Michael Peyper <https://github.com/mpeyper>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
// TypeScript Version: 3.0

import { ComponentType } from 'react';
export { act } from 'react-test-renderer';

export interface RenderHookOptions<P> {
    initialProps?: P;
    wrapper?: React.ComponentType;
}

export interface HookResult<R> {
    readonly current: R;
    readonly error: Error;
}

export interface RenderHookResult<P, R> {
    readonly result: HookResult<R>;
    readonly waitForNextUpdate: () => Promise<void>;
    readonly unmount: () => boolean;
    readonly rerender: (newProps?: P) => void;
}

/**
 * Renders a test component that will call the provided `callback`, including any hooks it calls, every time it renders.
 */
export function renderHook<P, R>(callback: (props: P) => R, options?: RenderHookOptions<P>): RenderHookResult<P, R>;

/**
 * Unmounts any rendered hooks rendered with `renderHook`, ensuring all effects have been flushed.
 */
export function cleanup(): Promise<void>;
```

### PR DESCRIPTION
**Pull request by sarahdayan**
fixes https://github.com/testing-library/react-hooks-testing-library/issues/246

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://react-hooks-testing-library.com/reference/api#waitforvaluetochange
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/testing-library__react-hooks/index.d.ts
Comment (mpeyper) [inline types/testing-library__react-hooks/index.d.ts]: `waitForNextUpdate` will also accept `WaitOptions` in `v3.2`

**Review by mpeyper — CHANGES_REQUESTED** [review]
Looks great, thanks! Just the minor change to the `waitForNextUpdate` signature and I'll approve this.

File: types/testing-library__react-hooks/index.d.ts
Comment (sarahdayan) [reply types/testing-library__react-hooks/index.d.ts]: Done.

**Review by mpeyper — APPROVED** [review]

**Review by kaplanburak — APPROVED** [review]

**Comment by syabro** [comment]
@sarahdayan could you please merge? :)

**Comment by sarahdayan** [comment]
@syabro I can't merge myself, a DefinitelyTyped maintainer must approve before so.

**Comment by mpeyper** [comment]
It appears like it's might have gone missing in the [tracking board](https://github.com/DefinitelyTyped/DefinitelyTyped/projects/3?card_filter_query=is%3Aopen) mentioned in the [FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped#ive-submitted-a-pull-request-how-long-until-it-is-merged).  I'm wondering if it's because the `Revision needed` tag was not removed after I approved the requested changes?

@RyanCavanaugh is the @typescript-bot misbehaving (the bot's bio says to bug you about it)?

**Pull request merged**
